ipa-core/src/helpers/transport/stream/mod.rs

```diff
@@ -42,6 +46,144 @@ pub type BoxBytesStream = Pin<Box<dyn BytesStream>>;
 //  * Avoiding an extra level of boxing in the production configuration using axum, since
 //    the axum body stream type is already a `Pin<Box<dyn HttpBody>>`.
 #[cfg(feature = "in-memory-infra")]
-pub type BodyStream = WrappedBoxBodyStream;
+type BodyStreamInner = WrappedBoxBodyStream;
 #[cfg(feature = "real-world-infra")]
-pub type BodyStream = WrappedAxumBodyStream;
+type BodyStreamInner = WrappedAxumBodyStream;
+
+/// Wrapper around [`BodyStreamInner`] that enforces checks relevant to both in-memory and
+/// real-world implementations.
+pub struct BodyStream {
+    inner: BodyStreamInner,
+}
+
+impl BodyStream {
+    pub fn new(bytes: Bytes) -> Self {
```

Comment:
since this is doing more than just initializing, should this be called `chunk` or `split`?